runtime/doc/provider.txt

```diff
@@ -88,6 +88,30 @@ the |:CheckHealth| command to diagnose your setup.
 		  save to a file or copy to the clipboard.
 
 ==============================================================================
+Ruby integration		    	      *provider-ruby*
+
+Nvim supports the Vim legacy |ruby-vim| interface via external Ruby
+interpreters connected via |RPC|.
+
+
+RUBY QUICKSTART ~
+
+To use Vim Ruby plugins with Nvim, just install the latest `neovim` RubyGem: >
+    $ gem install neovim
+
+
+RUBY PROVIDER CONFIGURATION ~
+						*g:loaded_ruby_provider*
+To disable Ruby support: >
+    let g:loaded_ruby_provider = 1
+
+
+TROUBLESHOOTING					*ruby-trouble*
+
+If you have trouble with a plugin that uses the `neovim` Ruby client, use the
+|:CheckHealth| command to diagnose your setup.
```

Comment:
To be honest I think we should remove `python-trouble` and move `:CheckHealth` somewhere else (don't need to do that in this PR though). In general we just want users to start reaching for `:CheckHealth` by habit. So there's no need to mention troubleshooting for any specific component.